common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookModel.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model.questbook;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.managers.Model;
+import com.wynntils.wynn.event.QuestBookReloadedEvent;
+import com.wynntils.wynn.model.container.ContainerContent;
+import com.wynntils.wynn.model.container.ScriptedContainerQuery;
+import com.wynntils.wynn.utils.InventoryUtils;
+import java.util.ArrayList;
+import java.util.List;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+
+public class QuestBookModel extends Model {
+    private static final int NEXT_PAGE_SLOT = 8;
+
+    private static List<QuestInfo> quests = List.of();
+    private static List<QuestInfo> newQuests;
+
+    public static List<QuestInfo> getQuests() {
+        return quests;
+    }
+
+    public static void queryQuestBook() {
+        ScriptedContainerQuery query = ScriptedContainerQuery.builder("Quest Book Query")
+                .useItemInHotbar(InventoryUtils.QUEST_BOOK_SLOT_NUM)
+                .matchTitle(getQuestBookTitle(1))
+                .processContainer(c -> processQuestBookPage(c, 1))
+                .clickOnSlotMatching(NEXT_PAGE_SLOT, Items.GOLDEN_SHOVEL, getNextPageButtonName(2))
+                .matchTitle(getQuestBookTitle(2))
+                .processContainer(c -> processQuestBookPage(c, 2))
+                .clickOnSlotMatching(NEXT_PAGE_SLOT, Items.GOLDEN_SHOVEL, getNextPageButtonName(3))
+                .matchTitle(getQuestBookTitle(3))
+                .processContainer(c -> processQuestBookPage(c, 3))
+                .clickOnSlotMatching(NEXT_PAGE_SLOT, Items.GOLDEN_SHOVEL, getNextPageButtonName(4))
+                .matchTitle(getQuestBookTitle(4))
+                .processContainer(c -> processQuestBookPage(c, 4))
+                .onError(msg -> WynntilsMod.warn("Error querying Quest Book:" + msg))
+                .build();
+
+        query.executeQuery();
```

Comment:
```suggestion
        QueryBuilder query = ScriptedContainerQuery.builder("Quest Book Query")
                .useItemInHotbar(InventoryUtils.QUEST_BOOK_SLOT_NUM)
                .onError(msg -> WynntilsMod.warn("Error querying Quest Book:" + msg));
        
        for (int page = 1; page <= QUEST_BOOK_SIZE; page++) {
                query.matchTitle(getQuestBookTitle(page))
                .processContainer(c -> processQuestBookPage(c, page))
                .clickOnSlotMatching(NEXT_PAGE_SLOT, Items.GOLDEN_SHOVEL, getNextPageButtonName(page + 1));
        }
                
        query.build()
                .executeQuery();
```
I was more aiming at reducing the magic numbers by doing something like the above. If we wanted to future proof it more we can read the book in the questbook's sidebar to get the number of pages in total (e.g. Math.ceil(137 / 42))

Reply:
(I coded the suggestion on my phone, please don't blindly commit)

Reply:
Ah, I see, you wanted to reduce duplication of code by introducing a loop. I guess that is fine, even though it is a bit unusual way of using a builder. If it is not, I'll fix the API so it works.  I'll take your suggestion and make sure it works, thanks!

Reply:
Fixed.

Fun fact: the new code is as many lines as the old. This one is hovering on the border where a loop is better than "unrolling" it. My hard rule is "never loop for 2 items", and my general rule is "only loop for 3 if it improves readability". Normally I'd agree that 4 items are enough to always be better with a loop, but this is really a border case, since the first container is different.